### DISCUSSION TITLE
Does Integrify only update it's target collections when one of it's master fields updates?

### DISCUSSION THREAD

**Discussion by narkeeso** [anishkny/integrify#77] [Q&A] [answered]
We have Integrify setup with one collection to try it out and I think one misconception was that Integrify would copy over fields when a new document was created on the target collection. Example:

```javascript
export const replicateUsers = integrify({
  rule: "REPLICATE_ATTRIBUTES",
  source: {
    collection: "users",
  },
  targets: [
    {
      collection: "projects",
      foreignKey: "user.id",
      attributeMapping: {
        username: "user.username",
        avatar: "user.avatar",
      },
    },
  ],
});
```

Are these attributes only replicated when the master collection fields are updated? Are we still responsible for creating the initial payload with intended replicated attributes?

**Comment by anishkny** [comment, marked as answer]
@narkeeso thanks for trying out `integrify`!

Yes, you are correct. The attributes are replicated from source to target only upon updating of the source document. The target document(s) need to exist for `integrify` to modify them.

Fundamentally, `REPLICATE_ATTRIBUTES` creates a Firestore `onUpdate` trigger, which is only called when a document exists and its value is changed (see [Cloud Firestore triggers docs](https://firebase.google.com/docs/functions/firestore-events)).

This is best suited for so called "one-to-many" relationships as explained in the [introductory blog post](https://dev.to/anishkny/---firestore-referential-integrity-via-triggers-kpb).

The reason for this is that `integrify` cannot predict how many target documents to create every time a new source document is created. In your example, every time a user document is created, how many project documents should be created? 0, 1, many? `REPLICATE_ATTRIBUTES` assumes that the required target documents already exist and updates their field values to follow the source document.

A workaround maybe to create a Firestore `onCreate` trigger on the `users` collection to create the required document(s) in the `projects` collection.

HTH

Edit: Created issue to try and clarify this point in the docs: https://github.com/anishkny/integrify/issues/78

**Reply by narkeeso** [reply]
Thanks for the clarification! Totally makes sense. We're loving integrify in our workflow and have found some really cool uses for it.

**Reply by anishkny** [reply]
@narkeeso Thats great to hear. If possible, would love to see some real world applications of `integrify` (assuming its okay with your organization, of course) - perhaps in blog post form or something?
